src/utils/shadow-css.ts

```diff
@@ -72,9 +72,19 @@ const _shadowDOMSelectorsRe = [/::shadow/g, /::content/g];
 
 const _selectorReSuffix = '([>\\s~+[.,{:][\\s\\S]*)?$';
 const _polyfillHostRe = /-shadowcsshost/gim;
-const _colonHostRe = /:host/gim;
-const _colonSlottedRe = /::slotted/gim;
-const _colonHostContextRe = /:host-context/gim;
+
+/**
+ * Little helper for generating a regex that will match a specified
+ * CSS selector when that selector is _not_ a part of a `@supports` rule.
+ *
```

Comment:
Can we add a little more detail here as to what we're checking/how the regex works? ATM, it's not super clear to me why we need both the negative and positive look-behinds. Is the idea something like:
```suggestion
 *
 * We build a regex that checks for either: 
 * 1. A string that does not start with `@supports`, but contains the provided selector.
 *    For a provided selector of ':host',  the following will match:
 *    - ':host' returns a match of ':host'
 *    - ':HOST' returns a match of ':HOST'
 *    - '.a:host' returns a match of ':host'
 *    For a provided selector of ':host', the following will not match:
 *    - '@supports :host' will not match, as the string begins with '@supports'
 *    - ':hostname' will not match, as we require a word boundary after the selector
 * 2. A string that matches the selector immediately following the provided selector. 
 *    For a provided selector of ':host',  the following will match:
 *    - '{:host' returns a match of ':host'
 *    - '{:HOST' returns a match of ':HOST'
 *    For a provided selector of ':host', the following will not match:
 *    - '{:hostname' will not match, as we require a word boundary after the selector
 * The regex captures the provided selector
 * 
```

Reply:
Added some more description and examples of what will and won't match